californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/LockstepEndpoint.java

```diff
@@ -316,6 +384,42 @@ public void check(Message message) {
 			for (Expectation<Message> expectation:expectations)
 				expectation.check(message);
 		}
+		
+		@Override
+		public void go() throws Exception {
+			if (null != multi) {
+				add(multi);
```

Comment:
couldn't this just be
````
if (null != multi) {
  multi.add(this);
  return;
}
````
and we drop the abstract `add(MultiMessageExpectation)` method?